messages/bitboxbase.proto

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// Should be sent every X seconds (TBD) unless the firmware already is busy with a command.
+message BitBoxBaseHeartBeatRequest {
+  enum StateCode {
+    IDLE = 0;
+    WORKING = 1;
+    ALERT_WARNING = 2;
+    ALERT_ERROR = 3;
+  };
+  StateCode state_code = 1;
+  enum DescriptionCode {
+    EMPTY = 0;
+    INITIAL_BLOCK_SYNC = 1;
+    DOWNLOAD_UPDATE = 2;
+    OUT_OF_DISK_SPACE = 3;
+  };
+  DescriptionCode description_code = 2;
+}
+
+// This will display the first 20 characters of the base32 encoded version of
+// the provided msg
+message BitBoxBaseConfirmPairingRequest {
+  bytes msg = 1;
+}
+
+// Optional fields can be represented by a "oneof" with only one field in it.
+// All fields are technically optional. But in reality the default value for the type will be set.
```

Comment:
interesting! the empty value could work though, right? empty ip or hostname is invalid.

Reply:
empty ip would be 0.0.0.0 and that is a valid ip. not sure if empty string is a valid hostname. But I guess not